planning/planning_scene_monitor/src/planning_scene_monitor.cpp

```diff
@@ -163,6 +165,11 @@ void planning_scene_monitor::PlanningSceneMonitor::initialize(const planning_sce
   moveit::tools::Profiler::ScopedStart prof_start;
   moveit::tools::Profiler::ScopedBlock prof_block("PlanningSceneMonitor::initialize");
 
+  // start our own spinner listening on our own callback_queue to become independent of any global callback queue
+  root_nh_.setCallbackQueue(&callback_queue_);
+  spinner_.reset(new ros::AsyncSpinner(1 /* threads */, &callback_queue_));
+  spinner_->start();
```

Comment:
@alainsanguinetti Could you have a look at this commit w.r.t. your pull-request #701?
Starting a spinner in the monitor obviously takes control out of the user's hands again.
Even so, afaics the PlanningSceneMonitor always used the global callback queue, right?
I don't see a way to provide the nodehandles the psm uses...

I suppose at least we should provide the user with a way to override the control flow from the outside.
tf/tf2 implement a similar scheme and they use an optional parameter `spin_thread` in the constructor, but this breaks ABI.
Is it enough to add the optional parameter in kinetic and leave indigo with the background thread?